Assets/Scripts/PlayerManager.cs

```diff
@@ -47,28 +48,46 @@ private void Start()
 
     private void Update()
     {
-        if (CanRest)
+        if (CanRest && Input.GetKeyDown(KeyCode.F))
         {
-            bool isResting = Input.GetKey(KeyCode.E);
-            UIManager.Instance.UpdateRestingText(isResting);
+            // Temporary Controls for Minigame
+            MinigameManager.Instance.LoadRandomMinigame();   
+        }
+    }
 
-            if (isResting)
-            {
-                HandleIncreaseStamina();
-            } 
-            else
-            {
-                // Temporary Controls for Minigame
-                if (Input.GetKeyDown(KeyCode.P))
-                {
-                    HandleTriggerStartMinigame();
-                }
-            }
+    private void FixedUpdate()
+    {
+        HandleDecreaseStamina();
+
+        float minDistance = distToClosestEnemy();
+
+        if (minDistance < dangerRadius)
+        {
+            HandleIncreaseStamina((dangerRadius - minDistance) / dangerRadius);
         }
-        else
+
+        // Temporary Controls for Minigame
+        if (Input.GetKeyDown(KeyCode.P))
+        {
+            HandleTriggerStartMinigame();
+        }
+    }
+
+    private float distToClosestEnemy()
+    {
+        GameObject[] enemies = GameObject.FindGameObjectsWithTag("Enemy");
```

Comment:
Just realized now, but maybe can we do like a [`Physics.OverlapSphere`](https://docs.unity3d.com/ScriptReference/Physics.OverlapSphere.html) instead? If we get a lot of enemies spawned in at some point, this will get a bit rough on performance.

Reply:
I'll add an issue for this, but for now the more expensive version will be fine since we have more important things to worry about.